File: src/main/java/com/ejo/game/scenes/TitleScene.java
package com.ejo.game.scenes;

import com.ejo.game.component.Background;
import com.ejo.game.math.Vector;
import com.ejo.game.component.Text;

import javax.swing.*;
import java.awt.*;

public class TitleScene extends Scene {

    private final JButton mainButton;
    private final Text title;

    private float sinStep = 0;

    private final Background background = new Background(100,10);

    public TitleScene() {
        super("Title Scene");
        this.mainButton = new JButton("Start!");
        this.title = new Text("Meteor Shower",new Font("Arial Black",Font.BOLD,50),Color.YELLOW);
        initStartButton();
        addSwingComponents(mainButton);
        background.initStarPositions();
    }

    @Override
    public void draw(Graphics2D graphics) {
        background.draw(graphics,Vector.NULL);
        drawSineTitle(graphics);
    }

    private void drawSineTitle(Graphics2D graphics) {
        title.draw(graphics,new Vector(getWindow().getSize().getX() / 2 - title.getSize().getX() / 2,getWindow().getSize().getY() / 2 + - title.getSize().getY() / 2 + 5 * Math.sin(sinStep)));
        sinStep += .075;
    }

    private void initStartButton() {
        mainButton.setSize(200, 50);
        mainButton.setLocation((int)(getWindow().getSize().getX() / 2 - mainButton.getWidth() / 2), (int)(getWindow().getSize().getY() / 2 - mainButton.getHeight() / 2) + 10);
        mainButton.setFont(new Font("Arial",Font.PLAIN,30));
        mainButton.addActionListener(action -> getWindow().setScene(new GameScene()));
    }

}

File: scene/TitleScene.py
from pygame import Color

from render.Text import Text, TextBounceHandler
from render.environment.Background import Background
from scene.GameScene import GameScene
from util.Vector import Vector
from render.Button import Button
from scene.Scene import Scene

class TitleScene(Scene):

    def __init__(self,window : 'Window'):
        super().__init__(window, "TitleScene")

        #Define Title Text & Bounce Handler
        self.text: Text = Text(self, Vector.null(), "Meteor Shower", "Arial Black", 70, Color(255, 255, 0))
        self.text.pos = (window.size * 0.5) - (self.text.get_vector_size() * 0.5) - Vector(0,50)
        self.title_bounce_handler: TextBounceHandler = TextBounceHandler(self.text, self.text.pos)

        #Define Start Button
        action = lambda: window.set_scene(GameScene(window))
        self.button: Button = Button(self,Vector.null(),Vector(250,100),Color(100,100,100),"Start!",action)
        self.button.pos = window.size * 0.5 - (self.button.size * 0.5) + Vector(0,75)

        #Define Starry Background
        self.background: Background = Background(self,window,50,5)

        #Add Elements
        self.drawables.append(self.background)
        self.drawables.append(self.text)
        self.drawables.append(self.button)

    def draw(self):
        super().draw()
        self.title_bounce_handler.update_pos()
